src/linker/Linker/MessageOrigin.cs

```diff
@@ -95,8 +105,15 @@ public override string ToString ()
 		public int CompareTo (MessageOrigin other)
 		{
 			if (MemberDefinition != null && other.MemberDefinition != null) {
-				return (MemberDefinition.DeclaringType?.Module?.Assembly?.Name?.Name, MemberDefinition.DeclaringType?.Name, MemberDefinition?.Name).CompareTo
+				int result = (MemberDefinition.DeclaringType?.Module?.Assembly?.Name?.Name, MemberDefinition.DeclaringType?.Name, MemberDefinition?.Name).CompareTo
```

Comment:
nit: does this need a `?` in every dereference? If the declaring type was non-null I think the rest is ok.

Reply:
Also, does this need a case for when the MemberDefinition is a type?

Reply:
I'm kind of split on this - I agree that it should not happen, but if it does it will hard crash the linker - with no really good reason (sorting warnings is not a critical feature anyway). I'd rather leave it as is for now.

Reply:
I fixed the type case (good catch)